_posts/2015-10-19-introduction_to_clojurescript.md

```diff
@@ -0,0 +1,269 @@
+
+```
+user=> (myfunction "hello world")
+"hello world"
+```
+
+In non-functional programming languages there are special "Operators" or keywords. 
+In Javascript, some commonly used operators are are `+ - == if`. In ClojureScript and other Lisp languages, there are no special operators. These are just regular functions. 
+
+If statements are functions
+
+```
+user=> (if true "do true stuff here" "do false stuff here")
+"do true stuff here"
+```
+
+Math operators are functions
+
+```
+user=> (+ 2 3)
+5
+user=> (* 2 3)
+6
+```
+
+For more great examples of [Javascript to ClojureScript synonyms checkout this site](https://kanaka.github.io/clojurescript/web/synonym.html)
+
+## Creating a Node.js - ClojureScript project
+Starting a ClojureScript project is simple. Leningen offers project templates that will get you up and running with a boilerplate project.
+Templates are a great resource to play around with and see other uses and configurations for ClojureScript projects. [Clojars.org has a collection of templates](https://clojars.org/search?q=template) and others can be found searching the web. For our project we'll be using a [Nodejs Figwheel project template](https://github.com/malyn/figwheel-node-template). At your terminal, run:
+
+```
+$ lein new figwheel-node hello-world
+$ cd hello-world
+$ npm install
+```
+
+This will create a new ClojureScript project in the directory `./hello-world`. Change to that directory and install NPM dependencies.
+
+### Points of interest
+
+* `package.json`
+This should be familiar from Node.js projects. Our NPM dependencies will be added here.
+
+* `project.clj`
+This file is the ClojureScript project configuration file. This is ClojureScripts version of package.json. This is where we configure Clojure dependencies and compilation targets. This file also contains project details like title and description.
+
+* `figwheel.js`
+This file is specific to Figweel projects. It is the bootstrap file for our project. It points Figwheel to our source code so that it can monitor it for updates. We will be running it with `node figwheel.js`
+
+* `./src/hello_world/core.cljs`
+This is our entry-point source file. This is where we will start the project. Think of it similar to a index.js file in a Node project.
+I've added comments below to explain what's going on.
+```
+;; This defines a namespace and necesscary dependencies for the current file
+(ns hello-world.core
+  (:require [cljs.nodejs :as nodejs]))
+
+(nodejs/enable-util-print!)
+
+;; The main function of the module
+;; it prints "Hello World!" to stdout
+(defn -main []
+  (println "Hello world!"))
+
+;; *main-cli-fn* is a semi-magic var that's used to set the entry
+;; *point for a node app
+(set! *main-cli-fn* -main)
+```
+
+### Running the project
+
+To execute the current project, open a terminal window and change directories to our hello_world project directory. Execute the following:
+```lein figwheel```
```

Comment:
might be worth adding (**not** `lein figwheel.js`) since that's what the shell auto-completes